### ISSUE TITLE
Try unloading all models before VAE if OOM

### ISSUE THREAD

**Issue by Jonseed** [kijai/ComfyUI-MochiWrapper#73]
I often get OOM before the VAE decode, but if I queue again, with latents from sampling still in memory, it works. After OOM, Comfy unloads all models. I think this clears up enough vram for VAE decode. 61 frames works with 9 tiles in VAE decode.

Here is my workflow:
https://gist.github.com/Jonseed/ce98489a981829ddd697fd498e2f3e22

**Comment by kijai** [comment]
You can also use the save latent node to save it to disk first with the Saved Latent -node, then you can try different decoding settings safely by loading it with Load Latent (it looks for the latest in your input folder)

**Comment by Jonseed** [comment]
I might also try adding an [unload all models node](https://github.com/SeanScripts/ComfyUI-Unload-Model) in-between the sampler and decode, just before the Mochi Decode node, so VAE has all the vram.

**Comment by Jonseed** [comment]
Yup, adding the `UnloadAllModels` node after the sampler before VAE decode worked for the whole queue, and I didn't get OOM. I'm on a 3060 12GB, btw.

**Comment by reydeljuego12345** [comment]
> Yup, adding the UnloadAllModels` node after the sampler before VAE decode worked for the whole queue, and I didn't get OOM. I'm on a 3060 12GB, btw.

I try this but simply don't work I have a 4090, 24 VRAM but for me is impossible to avoid OOM since few days, I don´t know what is happening.

**Comment by ggarra13** [comment]
@Jonseed I tried your workflow listed above and had the same issues you describe.  First an OOM and then the workflow worked when queued again.  I am on a 3080 RTX with 10GB, btw.

You said in this thread you added another UnloadAll node.  Would you be so kind to show that workflow?  Or is that the workflow shown above?

**Comment by Jonseed** [comment]
The unload nodes I use are from this repo: https://github.com/SeanScripts/ComfyUI-Unload-Model
They replicate the unload buttons in the ComfyUI but in node form. There is both an UnloadModel node, and UnloadAllModels node. I put the UnloadAllModels just before the VAE decoder (so the sampler outputs to it, and then it outputs from the UnloadAllModels to the VAE decoder), and haven't had nearly as many problems with OOM.

Here's an example workflow: https://gist.github.com/Jonseed/7ba98d34ef7684c25b73923f5578a844

**Comment by ggarra13** [comment]
@Jonseed Thanks!  It seems that with 10GB of VRAM I am limited to 720x480 resolution on your workflows.  Or... I have to run them in two passes.

**Comment by Jonseed** [comment]
yeah, 10GB is really barebones when it comes to video generation.
